src/hooks/useDeviceDetect.ts

```diff
@@ -0,0 +1,19 @@
+// utils/useDeviceDetect.js
+import React from 'react'
+
+export default function useDeviceDetect(): boolean {
+  const [isMobile, setMobile] = React.useState(false)
+
+  React.useEffect(() => {
+    const userAgent =
+      typeof window.navigator === 'undefined' ? '' : navigator.userAgent
+    const mobile = Boolean(
+      userAgent.match(
+        /Android|BlackBerry|iPhone|iPad|iPod|Opera Mini|IEMobile|WPDesktop/i
+      )
+    )
+    setMobile(mobile)
+  }, [])
```

Comment:
As this is a stateless check we can return a memoized value directly:

```suggestion
  return useMemo(() => {
    const userAgent =
      typeof window.navigator === 'undefined' ? '' : navigator.userAgent
    return Boolean(
      userAgent.match(
        /Android|BlackBerry|iPhone|iPad|iPod|Opera Mini|IEMobile|WPDesktop/i
      )
    )
  }, [])
```